ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py

```diff
@@ -2552,6 +2552,40 @@ def test_tensorflow_scalar_mul(
     )
 
 
+# segment_sum
+@handle_frontend_test(
+    fn_tree="tensorflow.math.segment_sum",
+    data=helpers.array_values(dtype=ivy.int32, shape=(5, 6), min_value=1, max_value=9),
+    segment_ids=helpers.array_values(
+        dtype=ivy.int32,
+        shape=(5,),
+        min_value=0,
+        max_value=4,
```

Comment:
Why are we hardcoding the dtype and shape here? Ideally, you should look into the `helpers.get_dtypes_and_values` function for generating the data for the test.

Reply:
According to the tensorflow docs https://www.tensorflow.org/api_docs/python/tf/math/segment_sum . segment_ids must be one of the following types: int32, int64. A 1-D tensor whose size is equal to the size of data's first dimension. That's why i hardcoded the shape and dtype here. Same goes for the min and max values of segment_ids, must be in a range not maximum than the shape of the segement_ids itself. The data can be of any dtype but for simplicity and to understand the data matrix while i was working i have choosen int.

Reply:
Thanks for explaning @pixel3user. `helpers.get_dtypes_and_values` has a `max_dims` parameter which you can set to 1 to generate only 1-dimensional arrays. Moreover, you can specify `dtype = ['int32', 'int64']` so that only these datatypes are used to generate the values inside the arrays. 

About the data, we should be as exhaustive as possible in the tests, so we should be generating all the dtypes and arbitrary shapes in our tests. Happy to help out if you run into any issues after updating these. :)